### PR TITLE
SDIT-642 Make consistent telemetry naming and exception technique

### PR DESCRIPTION
**Pull request by andymarke**
* This prior to making other domain use the same techniques
* Consistent telemetry names
* Switch to use `runCatching`

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesService.kt [hunk 1]
```diff
@@ -71,21 +71,22 @@ class ActivitiesService(
     }
 
   fun updateActivity(event: ScheduleDomainEvent) {
-    val telemetryMap = mutableMapOf("activityScheduleId" to event.additionalInformation.activityScheduleId.toString())
+    val activityScheduleId = event.additionalInformation.activityScheduleId
+    val telemetryMap = mutableMapOf("activityScheduleId" to activityScheduleId.toString())
 
     runCatching {
-      val activitySchedule = activitiesApiService.getActivitySchedule(event.additionalInformation.activityScheduleId)
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesService.kt:77]: switch order so mapping is the first thing checked to be consistent with creates

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesService.kt [hunk 2]
```diff
@@ -129,16 +130,14 @@ class ActivitiesService(
...
-          "allocationId" to allocation.id.toString(),
-          "offenderNo" to allocation.prisonerNumber,
-          "bookingId" to allocation.bookingId.toString(),
-        )
-
-        val nomisResponse = try {
+    val telemetryMap = mutableMapOf(
+      "allocationId" to allocationEvent.additionalInformation.allocationId.toString(),
+    )
+    runCatching {
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesService.kt:136]: switch to `runCatching` for consistency

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesService.kt [hunk 3]
```diff
@@ -147,31 +146,30 @@ class ActivitiesService(
...
-            "allocationId" to allocation.id.toString(),
-            "offenderNo" to allocation.prisonerNumber,
-            "bookingId" to allocation.bookingId.toString(),
-          )
+    val telemetryMap = mutableMapOf(
+      "allocationId" to allocationEvent.additionalInformation.allocationId.toString(),
+    )
 
-          val nomisResponse = try {
+    runCatching {
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesService.kt:167]: switch to `runCatching` for consistency

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesService.kt [hunk 4]
```diff
@@ -253,7 +251,7 @@ class ActivitiesService(
       ),
     ).also {
       telemetryClient.trackEvent(
-        "activity-retry-success",
+        "activity-create-mapping-retry-success",
         mapOf("activityScheduleId" to context.mapping.activityScheduleId.toString()),
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesService.kt:255]: more consistent with the the thing that is a success

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesServiceTest.kt [hunk 5]
```diff
@@ -177,6 +177,14 @@ internal class ActivitiesServiceTest {
 
     @Test
     fun `should throw and raise telemetry if cannot load Activity Schedule`() {
+      whenever(mappingService.getMappingGivenActivityScheduleId(anyLong())).thenReturn(
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesServiceTest.kt:180]: order has changed so mock now needs to be primed

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesServiceTest.kt [hunk 6]
```diff
@@ -208,7 +224,8 @@ internal class ActivitiesServiceTest {
       verify(telemetryClient).trackEvent(
         eq("activity-amend-failed"),
         check<Map<String, String>> {
-          assertThat(it).containsExactlyEntriesOf(mutableMapOf("activityScheduleId" to ACTIVITY_SCHEDULE_ID.toString()))
+          assertThat(it["nomisCourseActivityId"]).isEqualTo("$NOMIS_COURSE_ACTIVITY_ID")
+          assertThat(it["activityScheduleId"]).isEqualTo("$ACTIVITY_SCHEDULE_ID")
         },
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesServiceTest.kt:229]: we get further in this scenario so more is in telemetry

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesServiceTest.kt [hunk 7]
```diff
@@ -232,7 +249,6 @@ internal class ActivitiesServiceTest {
           assertThat(it).containsExactlyInAnyOrderEntriesOf(
             mapOf(
               "activityScheduleId" to ACTIVITY_SCHEDULE_ID.toString(),
-              "activityId" to ACTIVITY_ID.toString(),
             ),
           )
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesServiceTest.kt:253]: We don't get as far now so we have less in the telemetry (you win some, you lose some)

**Review by steverendell — APPROVED** [review]
LGTM

**Pull request merged**